src/game_inventory.cpp

```diff
@@ -2769,3 +2769,158 @@ std::pair<item_location, bool> game_menus::inv::unload( Character &you )
+                }
+                return std::string();
+            }, _( "DAMAGE" ) );
+
+            append_cell( []( const item_location & loc ) {
+                const itype *ammo = loc->is_ammo_container() ? loc->first_ammo().ammo_data() :
+                                    loc->ammo_data();
+                if( ammo ) {
+                    const damage_instance &dam = ammo->ammo->damage;
+                    return std::to_string( static_cast<int>( dam.empty() ? 0.0f : ( *dam.begin() ).res_pen ) );
+                }
+                return std::string();
+            }, _( "PIERCE" ) );
+        }
+
+        bool is_shown( const item_location &loc ) const override {
+            // todo: allow to reload a magazine/magazine well from a container pocket on the same item
+            if( loc.parent_item() == target ) {
+                return false;
+            }
+
+            if( !empty && loc->is_magazine() && !loc->ammo_remaining() ) {
+                return false;
+            }
+
+            std::vector<item_location> opts;
+            opts.emplace_back( target );
+
+            if( target->magazine_current() ) {
+                opts.emplace_back( target, const_cast<item *>( target->magazine_current() ) );
+            }
+
+            for( const item *mod : target->gunmods() ) {
+                item_location mod_loc( target, const_cast<item *>( mod ) );
+                opts.emplace_back( mod_loc );
+                if( mod->magazine_current() ) {
+                    opts.emplace_back( mod_loc, const_cast<item *>( mod->magazine_current() ) );
+                }
+            }
+
+            for( item_location &p : opts ) {
+                if( p->can_reload_with( *loc, false ) ) {
+                    return true;
+                }
+
+                if( ( loc->has_flag( flag_SPEEDLOADER ) && p->allows_speedloader( loc->typeId() ) &&
+                      loc->ammo_remaining() > 1 && p->ammo_remaining() < 1 ) && p->can_reload_with( *loc, false ) ) {
+                    return true;
+                }
+
+                if( you.can_reload( *p, loc.get_item() ) ) {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        // sort in order of move cost (ascending), then remaining ammo (descending) with empty magazines always last
+        bool sort_compare( const inventory_entry &lhs, const inventory_entry &rhs ) const override {
+            item_location left = lhs.any_item();
+            item_location right = rhs.any_item();
+
+            if( left->ammo_remaining() == 0 || right->ammo_remaining() == 0 ) {
+                return ( left->ammo_remaining() != 0 ) > ( right->ammo_remaining() != 0 );
+            }
+
+            if( left.obtain_cost( you ) != right.obtain_cost( you ) ) {
+                return left.obtain_cost( you ) < right.obtain_cost( you );
+            }
+
+            return left->ammo_remaining() > right->ammo_remaining();
```

Comment:
```suggestion
            if( left->ammo_remaining() != right->ammo_remaining() ) {
                return left->ammo_remaining() > right->ammo_remaining();
            }

            return inventory_selector_preset::sort_compare( lhs, rhs );
```

Stable sorting for identical items. Especially needed if you enable collation.